records/2023-01-26.bounce-rate-automated-emergency-brake.md

```diff
@@ -0,0 +1,83 @@
+
+## Proposed solution
+
+### Calculate service bounce rates
+Record the bounce status of every notification and calculate number of emails sent and the bounce rate for each service over the past [bounce-rate-window]
+
+### Notify a service when it is within [bounce-rate-warning-percentage] of the [bounce-rate-threshold]
+When a service is approaching the [bounce-rate-threshold], notify the service owner so they can take action to reduce their bounce rate.  The message should contain a link to the service's dashboard where they can see their bounce rate and take action to reduce it.
+
+#### Minimum volume threshold
+Low-volume senders won't have a large impact on the bounce rate, and notifying them directly may lead to many support calls. In order to limit this, only notify services that are sending more than [bounce-rate-service-volume-minimum] emails in the [bounce-rate-window].  This minimum could either be a fixed number or a percentage of the total number of emails sent by all services.
+
+### Suspend a service when it reaches the [bounce-rate-threshold]
+When a service reaches the [bounce-rate-threshold], suspend their service to prevent them from sending any more emails. Notify the service owner that their service has been suspended and provide them with a link to the service's dashboard where they can see their bounce rate and take action to reduce it.
+
+:question: _How to decide when it is appropriate to unsuspend a service?_
+
+### Add bounce rate information to the service dashboad
+Add the bounce rate to the service dashboard so that service owners can see their bounce rate and proactively read the guidance on it so they can prevent their bounce rate from rising in the first place.
+
+### Provide bounce rate guidance
+Provide guidance to service owners to explain bounce rate concepts and how they apply to their service.  Provide guidance on how services can keep their bounce rate low. 
+
+### bounce-rate-threshold calculation options
+When calculating a service's bounce rate, we can either consider them in isolation or in aggregate with other services.  The bounce rate is calculated as a percentage of the total number of emails sent.  If we consider a service in isolation, then the bounce rate will be calculated as a percentage of the total number of emails sent by that service.  If we consider a service in aggregate, then the bounce rate will be calculated as a percentage of the total number of emails sent by all services.
+
+#### Option 1: Consider a service in isolation
+Considering a service in isolation would mean calculating the bounce rate solely based on an individual services activity.  The benefit of this is that the calculation would be straight-forward and easy to understand by the service.  The downside is that even if a particular service has a high bounce rate, it may not have a large impact on the overall bounce rate. 
```

Comment:
Overall, if all services are below their 10% bounce rate, we can confidently say that the global bounce rate calculated on the total email should be below 10% as well, right?